src/Parsers/ParserDataType.cpp

```diff
@@ -27,7 +27,7 @@ class ParserDataTypeArgument : public IParserBase
     {
         ParserNestedTable nested_parser;
         ParserDataType data_type_parser;
-        ParserLiteral literal_parser;
+        ParserAllCollectionsOfLiterals literal_parser;
```

Comment:
After an update to the latest master, this change should be applied.

It is also worth noticing that ParserAllCollectionsOfLiterals does not parse literals (as in ParserLiteral), but only collections of literals.
```suggestion
        ParserAllCollectionsOfLiterals literal_parser(false);
```